### PR TITLE
New package: yacy-1.922

### PR DESCRIPTION
**Pull request by jason1987d**

### CODE REVIEW

File: srcpkgs/yacy/template [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+do_install() {
+vmkdir opt/yacy
+vcopy $XBPS_BUILDDIR/yacy-1.922/* opt/yacy
+}
+
+post_install() {
+vsv yacy
+}
```
Comment (ericonr) [inline srcpkgs/yacy/template:30]: indent with tabs

File: srcpkgs/yacy/template [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Template file for 'yacy'
+pkgname=yacy
+version=1.922
+revision=1
+archs="x86_64"
+depends="openjdk-jre"
```
Comment (ericonr) [inline srcpkgs/yacy/template:6]: why is it restriced to x86_64?

File: srcpkgs/yacy/template [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+maintainer="Jason Manley <jason@jasondavid.tv>"
+license="GPL-3.0-or-later"
+homepage="https://yacy.net"
+distfiles="http://latest.yacy.net/yacy_v${version}_20191013_9964.tar.gz"
+checksum=@99af963642dafaeab2657ad59315e7b7896bf4c1e7091f8f67e95e71802d44df
+
+system_accounts="yacy"
+yacy_homedir="/opt/yacy"
+yacy_shell="/bin/bash"
+make_dirs="/opt/yacy 0755 yacy yacy"
```
Comment (ericonr) [inline srcpkgs/yacy/template:17]: I believe it's preferred to have the homedir in `/var` (could be wrong)

**Comment by ericonr** [comment]
https://github.com/yacy/yacy_search_server#compile-from-source

File: srcpkgs/yacy/template (hunk 1 above)
Comment (jason1987d) [reply srcpkgs/yacy/template:29]: It works perfectly well in this case of not building from source. I removed the arch restriction too. I did do some testing with "ant clean all" in do_install but can't figure out what java package is needed in makedepends. It keeps erroring out trying to find java.

Comment (ericonr) [reply srcpkgs/yacy/template]: You need the actual development kit to build something with java, so add `openjdk` to `hostmakedepends`. You might need to set `JAVA_HOME` as well.

**Comment by ericonr** [comment]
Squash your commits when you're done, please.

File: srcpkgs/yacy/template (hunk 1 above)
Comment (jason1987d) [reply srcpkgs/yacy/template]: I did try with openjdk as a makedepends (in addition of course to apache-ant), but I'm not too familiar with how JAVA_HOME should be set here.

Comment (ericonr) [reply srcpkgs/yacy/template]: Take a look at how lightzone does it, for example.

File: srcpkgs/yacy/files/yacy/run [hunk 4]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+exec chpst -u yacy:yacy /opt/yacy/startYACY.sh -f
```
Comment (ericonr) [inline srcpkgs/yacy/files/yacy/run:2]: Are there other options users could pass to the start script? You can include a check for a conf file and source it if available. Several other services have this option.

Comment (jason1987d) [reply srcpkgs/yacy/files/yacy/run:2]: Maybe the -s switch here below has some potential. could set to ~/.yacy or something.

`/opt/yacy/startYACY.sh --help
startscript for YaCy on UNIX-like systems
Options
  -h, --help		show this help
  -t, --tail-log	show the output of "tail -f DATA/LOG/yacy00.log" after starting YaCy
  -l, --logging		save the output of YaCy to yacy.log
  -d, --debug		show the output of YaCy on the console and enable remote monitoring with JMX
  -f, --foreground	run as a foreground process, showing the output of YaCy on the console
  -p, --print-out	only print the command, which would be executed to start YaCy
  -s, --startup [data-path] start YaCy using the specified data folder path, relative to the current user home
  -g, --gui		start a gui for YaCy
[jason@box ~]$ ls /opt/yacy
addon    build.properties  DATA                 gpl.txt                        killYACY.sh  libbuild  reconfigureYACY.sh  startYACY.bat        stopYACY.bat                     updateYACY.sh
AUTHORS  build.xml         defaults             htroot                         langdetect   locales   skins               startYACY_debug.bat  stopYACY.sh                      yacy.yellow
bin      COPYRIGHT         getWin32MaxHeap.bat  installYaCyWindowsService.bat  lib          NOTICE    source              startYACY.sh         uninstallYaCyWindowsService.bat
[jason@box ~]$ ls /opt/yacy/DATA/
ARCHIVE  CLASSIFICATION  DICTIONARIES  HTCACHE  HTDOCS  INDEX  LISTS  LOG  RELEASE  SETTINGS  SKINS  SURROGATES  WORK  yacy.running`

Comment (jason1987d) [reply srcpkgs/yacy/files/yacy/run:2]: `/opt/yacy/startYACY.sh --help
startscript for YaCy on UNIX-like systems
Options
  -h, --help		show this help
  -t, --tail-log	show the output of "tail -f DATA/LOG/yacy00.log" after starting YaCy
  -l, --logging		save the output of YaCy to yacy.log
  -d, --debug		show the output of YaCy on the console and enable remote monitoring with JMX
  -f, --foreground	run as a foreground process, showing the output of YaCy on the console
  -p, --print-out	only print the command, which would be executed to start YaCy
  -s, --startup [data-path] start YaCy using the specified data folder path, relative to the current user home
  -g, --gui		start a gui for YaCy

[jason@box ~]$ ls /opt/yacy
addon    build.properties  DATA                 gpl.txt                        killYACY.sh  libbuild  reconfigureYACY.sh  startYACY.bat        stopYACY.bat                     updateYACY.sh
AUTHORS  build.xml         defaults             htroot                         langdetect   locales   skins               startYACY_debug.bat  stopYACY.sh                      yacy.yellow
bin      COPYRIGHT         getWin32MaxHeap.bat  installYaCyWindowsService.bat  lib          NOTICE    source              startYACY.sh         uninstallYaCyWindowsService.bat

[jason@box ~]$ ls /opt/yacy/DATA/
ARCHIVE  CLASSIFICATION  DICTIONARIES  HTCACHE  HTDOCS  INDEX  LISTS  LOG  RELEASE  SETTINGS  SKINS  SURROGATES  WORK  yacy.running`

**Comment by fosslinux** [comment]
Remove the google-earth-pro stuff from this PR too, please.

**Comment by jason1987d** [comment]
How would I remove those commits?
git rebase -i <commit id> 
???

**Comment by jason1987d** [comment]
> This should be compiled from source. I don't see any reason for this not to be compiled from source...
> 
> Resources:
> 
> https://github.com/yacy/yacy_search_server#compile-from-source
> https://github.com/yacy/yacy_search_server/archive/Release_1.92.tar.gz

I am testing building from source and will have this PR up to date with that shortly. /var/lib may be the better place for this as well (I think similar to packages like transmission).

**Comment by jason1987d** [comment]
I made some updates such as to the path. I hope I didn't botch the commit trail trying to remove my mistaken google-earth-pro commits.

**Comment by fosslinux** [comment]
Ok, looking better. I'll test this soon ish. The merge commit should not be kept, use rebase instead. Not sure how to remove that though

**Comment by jason1987d** [comment]
Anything new on this? Do you want me to close out and start a new pull request since this one has dirty commit history? Thank you.

**Comment by ericonr** [comment]
@jason1987d you can run `git rebase -i HEAD~9` (where 9 is the number of commits you have here) and replace `pick` with `squash` for all of them. Then you just need to fix the final commit message and force push the branch here.

**Comment by jason1987d** [comment]
> @jason1987d you can run `git rebase -i HEAD~9` (where 9 is the number of commits you have here) and replace `pick` with `squash` for all of them. Then you just need to fix the final commit message and force push the branch here.

That did not work. 
fatal: invalid upstream 'HEAD-9'

**Comment by ericonr** [comment]
It's `~`, not `-`

**Comment by jason1987d** [comment]
Didn't see that ~ vs - at first, thank you. I have done plenty of squashing, but nothing seems to change here on the commits in this pull request.

**Comment by ericonr** [comment]
Not exactly. Can you check if `git log` shows all the commits that are in https://github.com/void-linux/void-packages/pull/23695/commits ? If not, you can run `git push -f origin master`.

**Comment by jason1987d** [comment]
Looking at git log does show these commits.

**Comment by jason1987d** [comment]
Each attempt at a squash just mangles the main template file with random git gibberish. What am I doing wrong here? Is it okay if I close this pull request and start fresh with just the one commit now that we've discussed and fixed what was actually wrong with the template?

**Comment by ericonr** [comment]
Sure.

**Comment by Chocimier** [comment]
> Is it okay if I close this pull request and start fresh with just the one commit

It is preferred to make changes as for new PR, then `git push --force FORK NEW_BRANCH:BRANCH_OF_EXISTING_PR`

**Pull request closed without merging**